ocaml/libs/clock/test_timer.ml

```diff
@@ -26,11 +26,16 @@ let test_timer_remaining =
   let () =
     match Timer.remaining timer with
     | Expired t ->
-        Test.fail_reportf
-          "Expected to have spare time, but got excess: %a. Duration: %a, \
-           actual: %a, timer: %a"
-          Mtime.Span.pp t Mtime.Span.pp duration Mtime.Span.pp actual Timer.pp
-          timer
+        (* OS timer may not be accurate for very short sleeps,
+           or the system might be busy.
+           Skip the strict test on very short durations, we'll still test this on the 100ms+ ones.
+        *)
+        if half >= 0.05 then
```

Comment:
This code has a silent `else` branch. I think it would be more obvious to use `| Expired t when half < 0.05 ->` and another `| Expired t ->` where we fail.